cmake/kokkos_functions.cmake

```diff
@@ -954,6 +954,113 @@ int main()
   set(${_VAR} ${_RET} CACHE STRING "CXX compiler supports building CUDA")
 endfunction()
 
+# this function checks if the compiler supports specific flags. No-op for CMake < 3.19:
+#
+#       SILENT    --> check without printing and erroring out (default=OFF)
+#       LANGUAGE  --> compile language (required)
+#       FLAGS     --> list of compiler flags
+#       OPTIONS   --> list of options that get printed in case of failure (optional)
+#       RESULT    --> bool with test result (output, optional)
+#
+function(kokkos_check_compiler_flags)
+  cmake_parse_arguments(INP "SILENT" "LANGUAGE;RESULT" "FLAGS;OPTIONS" ${ARGN})
+
+  if(NOT INP_LANGUAGE)
+    message(FATAL_ERROR "'kokkos_check_compiler_flags' requires a LANGUAGE to be passed.")
+  endif()
+
+  #if no check is performed return true
+  if(DEFINED INP_RESULT)
+    set(${INP_RESULT} ON PARENT_SCOPE)
+  endif()
+
+  set(CMAKE_REQUIRED_QUIET ON)
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.19)
```

Comment:
I'm not a fan of different behavior on different CMake versions (I know we do it elsewhere, not a fan of it there either). I'd rather bump the cmake version though of course that's complicated on its own

Reply:
I guess this is a longer discussion? https://github.com/kokkos/kokkos/pull/7891#discussion_r1996022565